### PR TITLE
Extract resource type codes into a dedicated class

### PR DESCRIPTION
**Pull request by lgonczar**
... so that each of them is defined once.

### CODE REVIEW

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ResourceTypeCodes.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package uk.nhs.adaptors.gp2gp.ehr.mapper;
+
+public final class ResourceTypeCodes {
+
+    private ResourceTypeCodes() {
+        // hiding the constructor
+    }
```
Comment (msolecka) [inline service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ResourceTypeCodes.java:7]:
```suggestion
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public final class ResourceTypeCodes {

```
lombok annotation could be used instead

**Review by mattd-kainos — DISMISSED** [review]

**Review by mattd-kainos — APPROVED** [review]

**Review by msolecka — APPROVED** [review]

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ResourceTypeCodes.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    public static final String TOPIC_LIST_CODE = "25851000000105";
+    public static final String CATEGORY_LIST_CODE = "24781000000107";
+    public static final String BLOOD_PRESSURE_READING_CODE = "163020007";
+    public static final String ARTERIAL_BLOOD_PRESSURE_CODE = "386534000";
+    public static final String BLOOD_PRESSURE_CODE = "75367002";
+    public static final String STANDING_BLOOD_PRESSURE_CODE = "163034007";
+    public static final String SITTING_BLOOD_PRESSURE_CODE = "163035008";
+    public static final String LAYING_BLOOD_PRESSURE_CODE = "163033001";
+    public static final String ENCOUNTER_LIST_CODE = "1149501000000101";
+    public static final String NARRATIVE_STATEMENT_CODE = "37331000000100";
```
Comment (bartek-sarul) [inline service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ResourceTypeCodes.java:20]: `NARRATIVE_STATEMENT_CODE` is the same as `COMMENT_NOTE_CODE`
I know it's the same as before your change but can you confirm this is ok?

**Review by bartek-sarul — CHANGES_REQUESTED** [review]

**Comment by bartek-sarul** [comment]
Branch name is incorrect. Should be `feature/NIAD-<ticket number>-<description>` e.g. `feature/NIAD-123-something`

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ResourceTypeCodes.java (hunk 2 above)
Comment (lgonczar) [reply service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ResourceTypeCodes.java:20]: Both codes are used for checking if the observation is a narrative. I'll remove one of them, then.

**Comment by lgonczar** [comment]
> Branch name is incorrect. Should be `feature/NIAD-<ticket number>-<description>` e.g. `feature/NIAD-123-something`

This refactoring bit hasn't been done as part of any story - I didn't want to mix it with my ticket-specific changes in a single pull request.

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ResourceTypeCodes.java (hunk 2 above)
Comment (lgonczar) [reply service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ResourceTypeCodes.java:20]: done

**Comment by mattd-kainos** [comment]
@lgonczar can we get the conflicts resolved so we can merge this?

**Comment by mattd-kainos** [comment]
Never mind, it was closed as I wrote my comment.

**Pull request closed without merging**